scanpipe/pipes/glc.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# http://nexb.com and https://github.com/nexB/scancode.io
+# The ScanCode.io software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode.io is provided as-is without warranties.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# Data Generated with ScanCode.io is provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+# ScanCode.io should be considered or used as legal advice. Consult an Attorney
+# for any legal advice.
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/nexB/scancode.io for support and download.
+
+
+from LicenseClassifier.classifier import LicenseClassifier
+
+from scanpipe.pipes import scancode
+
+
+def scan_directory(location):
+    """
+    Run a license and copyright scan on directory at `location`,
+    using golicense-classifier.
+    """
+    classifier = LicenseClassifier()
+    result = classifier.scan_directory(location)
+    return result
+
+
+def scan_file_for_license(location):
+    """
+    Run a license and copyright scan on provided `location`,
+    using golicense-classifier.
+    """
+    classifier = LicenseClassifier()
+    result = classifier.scan_file(location)
+    return result
+
+
+def update_codebase_resources(project, scan_data):
+    """
+    Update CodebaseResource model with scan results from golicense-classifier
+    """
+    for resource in project.codebaseresources.no_status():
+        scan_result = next(
```

Comment:
This needs to be simplified. 
Also performance wise, it's not great to have to iterate over the full results set for each resource. 
What about making an index of those results using the path as key instead?

Reply:
This is something that pertains to the output structure we use for the result. Can you have a look at #231?

Reply:
Please see https://github.com/nexB/scancode.io/pull/226#issuecomment-879912975 
This nested loop is not needed and is not related to the format documentation of #231 . If you work one file ta a time that's a non-issue.